pkg/collector/python/init.go

```diff
@@ -304,7 +304,8 @@ func pathToBinary(name string, ignoreErrors bool) (string, error) {
 
 func resolvePythonExecPath(ignoreErrors bool) (string, error) {
 	// Since the install location can be set by the user on Windows we use relative import
-	if runtime.GOOS == "windows" {
+	// MacOS runners are going to build the agent at a custom directory hence we also use relative import
+	if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
```

Comment:
Would there be any reasons why this heuristic isn't generally applicable such that we could remove the conditional here altogether?

Reply:
It might or might not be useful but it makes the code simpler by not making this behaviour a special cae.

Reply:
But adding support to linux would allow the removal of the `--python-home-3` I think from omnibus 🤔

Reply:
Would that be a problem? The same line inside omnibus is used on both Linux and macOS anyway:
https://github.com/DataDog/datadog-agent/blob/bb84cf8166e2d1623f455231c93fa483e60ac470/omnibus/config/software/datadog-agent.rb#L86

Reply:
But yea I'll remove it all together, I don't see how it can end up badly

Reply:
do we even needs `--python-home-2` 😮 ? Just in case I'll clean these in a separate PR

Reply:
We don't, I agree with the cleanup on a separate PR 👍